### PR TITLE
Add CMakeLists.txt and build instructions

### PR DESCRIPTION
**Pull request by angristan**
Makes it easier to run the demo for newcomers!

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,30 @@
-cmake_minimum_required(VERSION 3.13)
-PROJECT (dejavu_cpp_port)
+#./CMakeLists.txt
```
Comment (angristan) [inline CMakeLists.txt:1]:
```suggestion
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
+cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
+cmake_policy(VERSION 3.2)
+
+project(dejavu_cpp_port
+    VERSION "1.0.0"
```
Comment (angristan) [inline CMakeLists.txt:6]: There is no published version: https://github.com/salsowelim/dejavu_cpp_port/tags

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
+    DESCRIPTION "C++ port of dejavu lib"
+)
 find_package(OpenCV REQUIRED)
 
-set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
-add_executable(fingerprint fingerprint.cpp fingerprint.h)
+set(CMAKE_CXX_STANDARD 17)
```
Comment (angristan) [inline CMakeLists.txt:11]: If you change that also update the README which indicates C++14

File: CMakeLists.txt (hunk 2 above)
Comment (angristan) [inline CMakeLists.txt:6]:
```suggestion
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,8 +1,30 @@
...
+    "${CMAKE_CURRENT_SOURCE_DIR}/include"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src"
+)
+
+target_link_libraries(${PROJECT_NAME} ${OpenCV_LIBS})
+
+target_include_directories(${PROJECT_NAME} PUBLIC ${INCLUDES})
+
+
+
```
Comment (angristan) [inline CMakeLists.txt:29]: Remove newlines

**Review by angristan — COMMENTED** [review]
Did you remove the mp3 file by mistake? @Vagahbond

**Comment by Vagahbond** [comment]
No, it's not healthy to have bins in a repository.

**Comment by angristan** [comment]
It's not a binary fin

File: CMakeLists.txt (hunk 2 above)
Comment (Vagahbond) [reply CMakeLists.txt:6]: fair enough

File: CMakeLists.txt (hunk 3 above)
Comment (Vagahbond) [reply CMakeLists.txt:11]: yes

**Pull request closed without merging**